### ISSUE TITLE
Failed to initialize CoreCLR, HRESULT: 0x80004005 

### ISSUE THREAD

**Issue by prafullbhosale** [dotnet/cli#1488]
I have a project with a tool defined in the project.json's "tools" key. 
dotnet restore is able to find the tool and restore it succesfully. 

However when I try to invoke the tool, I get the error:
### `Failed to initialize CoreCLR, HRESULT: 0x80004005`
- [project.json](https://github.com/aspnet/Scaffolding/blob/pb/dotnettool/src/dotnet-aspnet-codegenerator/project.json) of the tool.
- [project.json](https://github.com/prafullbhosale/scaffoldingsample/blob/master/SampleWebApp/src/SampleWebApp/project.json) of the project that uses the tool. 

What am I missing here?

**Comment by piotrpMSFT** [comment]
Does your environment have a DOTNET_HOME directory set? Does this repro without it? What is the associated `dotnet --version`?

**Comment by 18583690825** [comment]
#I got the same erro on CentOS6.5 .I updated my glibc & gcc. And i got the Binaries of Cli from #https://github.com/dotnet/cli and added it to the PATH. But when i run 'dotnet',it shows 'Failed to #initialize CoreCLR, HRESULT: 0x80131500'
#please help...

[root@Cent65-Core down]# mkdir demo
[root@Cent65-Core down]# cd demo/
[root@Cent65-Core demo]# dotnet new
Failed to initialize CoreCLR, HRESULT: 0x80131500
[root@Cent65-Core demo]# dotnet
Failed to initialize CoreCLR, HRESULT: 0x80131500
[root@Cent65-Core demo]# dotnet -version
Failed to initialize CoreCLR, HRESULT: 0x80131500

#And my dnx run faild.It has any info. I'm sure it is running ,but something wrong and it's dropped out..


[root@Cent65-Core demo]# dnvm list

Active Version              Runtime Architecture OperatingSystem Alias
------ -------              ------- ------------ --------------- -----
  *    1.0.0-rc1-update1    coreclr x64          linux           default

#Created a 'project.json' & 'Program.cs' manual . Then...↓
[root@Cent65-Core demo]# dnu restore
#nothing happend..↑
[root@Cent65-Core demo]# dnx run
#nothing happend..↑

**Comment by prafullbhosale** [comment]
@piotrpMSFT 
I don't have this issue anymore with dotnet cli 1.0.0-beta-001489. 

With the earlier version, I did try with and without dotnet_home env variable and still had the issue

The only thing I observed (though I am not sure that it could be the cause) was that the tool was built with using older dotnet-cli version (a day older) than what the project was built with. 

I will try to see if I can get this to repro again.

**Comment by colemickens** [comment]
I can repro. I used this Dockerfile: https://raw.githubusercontent.com/colemickens/polykube/dev/dotnet/dockerfiles/aspnet-base/Dockerfile

Cloned `dotnet/cli-samples`.

Ran `dotnet restore` and `COREHOST_TRACE=1 dotnet run`: https://gist.github.com/colemickens/133e434131961e6fe28178a35ed03696

`Failed to initialize CoreCLR, HRESULT: 0x80004005`

**Comment by hstarorg** [comment]
Try ``yum install icu``

**Comment by jamesqo** [comment]
I seem to be getting this error too with the RC2/preview1 build on Windows. Not sure if this has anything to do with it, but I used `xcopy` to replace my versions of `mscorlib.dll` and `mscorlib.ni.dll` as I am attempting to measure the perf impact of some changes to coreclr.

Here is the output with `COREHOST_TRACE=1`: https://gist.github.com/anonymous/5f1ea6c1161a1850fc84cdabcef07f02

**Comment by justinjstark** [comment]
I just installed on Ubuntu 16.04 using the directions at https://www.microsoft.com/net/core#ubuntu except I dropped it in /opt/dotnet instead of ~/dotnet.

```
justinjstark@X230:~/test$ dotnet
Usage: dotnet [--help | app.dll]
justinjstark@X230:~/test$ dotnet --version
Failed to initialize CoreCLR, HRESULT: 0x80131500
justinjstark@X230:~/test$ dotnet --help
Failed to initialize CoreCLR, HRESULT: 0x80131500
justinjstark@X230:~/test$ dotnet new
Failed to initialize CoreCLR, HRESULT: 0x80131500
```

**Comment by justinjstark** [comment]
I just installed on Ubuntu 16.04 using the directions at https://www.microsoft.com/net/core#ubuntu except I dropped it in /opt/dotnet instead of ~/dotnet.

```
justinjstark@X230:~/test$ dotnet
Usage: dotnet [--help | app.dll]
justinjstark@X230:~/test$ dotnet --version
Failed to initialize CoreCLR, HRESULT: 0x80131500
justinjstark@X230:~/test$ dotnet --help
Failed to initialize CoreCLR, HRESULT: 0x80131500
justinjstark@X230:~/test$ dotnet new
Failed to initialize CoreCLR, HRESULT: 0x80131500
```

**Comment by colemickens** [comment]
Xenial isn't supported yet.

**Comment by justinjstark** [comment]
@colemickens Got it. Thanks.

**Comment by livarcocc** [comment]
This does not seem to be an issue anymore.

**Comment by jgomo3** [comment]
Not. I'm still having it. I'm using ElementaryOS Loki (it is actually Ubuntu 16.04), installed:

* `libicu55`
* libicu-dev

And this error is showing when running the `build.sh` script from [omnisharp](https://github.com/OmniSharp/omnisharp-roslyn), which is now using `dotnet-cli`.

**Comment by kamilosxd678** [comment]
I don't know why this is closed, because .NET Core stability is some kind of a joke.

http://pastebin.com/raw/tdSdXDKa

I have to issue _dotnet run_ command **37 TIMES** before it actually works. **THIRTY SEVEN GODDAMN TIMES** OR MORE!

**Comment by geeooff** [comment]
I tried all RTM versions on my Debian 8.6 (jessie) server, but my pre-compiled ASP.NET Core MVC webapp failed with this exact error.

My webapp is running as a _systemd service unit_ that call `dotnet my.dll --server.urls http://localhost:xxxx` with specific environment variables, and systemd is constantly trying to run and rerun the unit that fail constantly.

After several tries, sometimes, the webapp pass the startup initialization, but on first web client request an `OutOfMemoryException` is thrown, related to the lambda expression compiler used by entityframework apis.

This error, `Failed to initialize CoreCLR`, seems to be related to `OutOfMemoryException` exceptions.

Actually, this server has 4GB of RAM.
On another server using Ubuntu Server LTS, with 8GB of RAM, no exceptions. The webapp run fine.

On both server, when I use `htop` command, dotnet process has allocated 18.6GB of virtual memory for each child process. I'm not a linux specialist, especially with virtual memory, but is it really normal ?

**Comment by piotrpMSFT** [comment]
Reopening as there is obviously another issue here. @schellap @gkhanna79 can you guys take a look at these OOMs? What data do we need to make this experience better?

**Comment by gkhanna79** [comment]
@geeooff Is this a deterministic repro on Debian8.6? If so, can you please share the steps to setup a repro at our end? This will enable @kouvel to take a look at this further.

Also, if the repro is deterministic, have you had a chance to run it under gdb/lldb and break at when exception is thrown? If so, can you please share that dump or callstack?

CC @janvorli

**Comment by janvorli** [comment]
@kamilosxd678 I've just seen your previous post, I am sorry for missing it when you have posted it few weeks ago. What is the hardware configuration of your machine? 
Getting `Failed to initialize CoreCLR, HRESULT: 0x80004005` means that the initialization has failed at very early stages and I have always seen this error in the past when there was a mismatch between the libcoreclr.so and mscorlib.dll / mscorlib.ni.dl / System.Private.CorLib.dll / System.Private.ni.CorLib.dll  - e.g. they weren't built from the exactly the same sources  or libcoreclr.so was debug build and the other libs release or vice versa.
Have you built coreclr yourself or have you used a dotnet cli package that you've downloaded untouched?

**Comment by kamilosxd678** [comment]
@janvorli I have basically bare-bone installation from https://www.microsoft.com/net/core#linuxdebian this tutorial. Even basic example hello world app has these problems and my server is not some weak virtual machine but middle-grade dedicated server (32 GB RAM, Xeon E5, SSD drive) and I still have problems with it. I would agree that this is strictly bad installation problem, but it works randomly after throwing 0x80004005 and Out of memory messages.

**Comment by janvorli** [comment]
@kamilosxd678 the out of memory is really strange on a machine with so much memory. Is it possible that you have cgroups installed and that it limits the amount of memory available for the .NET SDK? 
Anyways, would you be willing to run the dotnet run in a little different way and under strace and get me the log when it fails? I may be able to dig the real culprit from that.

What I would need is the output of:
`strace dotnet exec bin/Debug/netcoreapp1.0/WebApplication2.dll`

**Comment by schellap** [comment]
I have seen something similar before https://github.com/dotnet/cli/issues/652, with the server GC switch. Could you also try toggling the GC to be server or not in the project file? Or in the runtimeconfig.json.

```
  "runtimeOptions": {
    "configProperties": {
      "System.GC.Server": false
    }
  },
```

**Comment by kamilosxd678** [comment]
@janvorli Here it is http://cx453.net/download/strace.log

**Comment by geeooff** [comment]
@gkhanna79 I will try to build another public project to test, this one is just too big to reproduce, and source code can't be shared unfortunately. For the moment, my app was built on a VSTS Build Server (Azure Hosted Agent) using `dotnet publish` command. Technically this is the only .NET Core project this machine had launched. Can this be related to Debian vs Ubuntu differences ? On my test laptop using Ubuntu, I didn't had problem too. I don't have any other Debian distro to test, unfortunately.

@janvorli This server is using the latest binaries available at [https://www.microsoft.com/net/core#linuxdebian](url), like @kamilosxd678 used. I will try to produce a trace too.

@schellap I will try this too. Thanks.

**Comment by gkhanna79** [comment]
> Can this be related to Debian vs Ubuntu differences ? 

It is possible and this is what I would like to isolate. If you can get a crash dump and share that, that may help too.

**Comment by janvorli** [comment]
@kamilosxd678 thanks a lot for the log. It has revealed one substantial anomaly that I believe causes the problem you are seeing, but I need to fully understand what's exactly going on. 
The first part of the issue is that at the startup, we are trying to reserve a block of continuous memory at least 128MB and at most 2GB large that's located within 2GB range from the location of the libcoreclr.so in memory. If we succeed, we use it for allocating memory for executable code. See https://github.com/dotnet/coreclr/pull/2205 for some details on the performance reasons for this allocator.
In your case, this fails to find even the 128MB region. We should still work fine even in this case, only the performance would be lower. But in the log, I can see that we are attempting to find a 64kB block of memory in certain virtual address range (of 2GB) and we also fail here and that likely leads to the exit with failure (it is hard to be sure just from the evidence from the log, but I am pretty confident it is the case). Given the fact that the blocks that it tries to reserve are 64kB long, I guess it comes from the JIT memory allocator.
It looks like the virtual memory is very fragmented or something like that. I think I can get some more info / evidence if you ran the same thing under gdb with a breakpoint set to some early place in the runtime, but I need to check what symbol would work for this and be available in the release build.

**Comment by geeooff** [comment]
@janvorli here's my log file: [https://dl.dropboxusercontent.com/u/7169701/GitHub/dotnet/strace_2016-11-23_1.log](url). I have renamed some parts of it, for obvious privacy reasons. I hope it will not bother you.

**Comment by janvorli** [comment]
@geeooff it shows a similar pattern. I cannot see the attempt to allocate memory for the executable memory allocator, but the pattern of looping over the large range of memory by 64k increments and failing after that is the same.

**Comment by janvorli** [comment]
@kamilosxd678 could you please run 
`gdb --args dotnet exec bin/Debug/netcoreapp1.0/WebApplication2.dll`
then set breakpoint:
`b ExecutableMemoryAllocator::TryReserveInitialMemory`
answer "y" to "Make breakpoint pending on future shared library load? (y or [n])"
and then run the app:
`r`
When you hit the breakpoint, run
`info proc map`
and get me the output. This shows the layout of the memory at the point where we are trying to reserve memory for the executable allocator.

**Comment by kamilosxd678** [comment]
@janvorli Sure thing
http://pastebin.com/LeNYCGwE

**Comment by geeooff** [comment]
@janvorli I did the same thing, log file is [here](https://dl.dropboxusercontent.com/u/7169701/GitHub/dotnet/gdb_2016-11-23_breakpoint.log).

**Comment by janvorli** [comment]
@kamilosxd678, @geeooff  when you use `c` debugger command after dumping the map, do you hit the problem? The thing is that in both of the logs you have provided, the memory is free in the range where the executable allocator would try to reserve it, so I wonder if that's caused by the fact we are running under GDB or if it is an indicator of the fact that the mmap for some reason behaves differently on your Debian and doesn't attempt to honor the requested address.
If the issue still happens under gdb, can you also set the following breakpoint:
`b ClrVirtualAllocWithinRange`
and see if after hitting the first one, we hit this one too? That would confirm my theory on where the attempts to 64k allocations at specific addresses are coming from.

**Comment by geeooff** [comment]
@kamilosxd678 I hit `Failed to initialize CoreCLR, HRESULT: 0x80004005` error after continuing from breakpoint. I tried to set your two breakpoints, and execution stopped on both breakpoints before crash. Here's the [log file](https://dl.dropboxusercontent.com/u/7169701/GitHub/dotnet/gdb_2016-11-23_two-breakpoints.log).

**Comment by kamilosxd678** [comment]
@geeooff I think you wanted to respond to @janvorli 

@janvorli So far every 10 times I continued program in gdb it was working ok :-( But without it - it still crashes

**Comment by janvorli** [comment]
Ok, then it seems that mmap is behaving in a strange way on Debian. I am installing debian 8.6 into a VM right now so that I can try to repro and debug the issue locally.

**Comment by kouvel** [comment]
I tried some simple tests on Debian 8.6, wasn't able to repro with server GC on or off. My VM has 4 GB RAM and I just installed with the default settings for relevant options. Were there any significant config changes made to your machines apart from the default settings?

**Comment by geeooff** [comment]
@kouvel It's a physical server, with a tiny CPU (an Intel Atom N2800, 2 cores, 4 threads), provided by OVH french ISP. 

```
vendor_id       : GenuineIntel
cpu family      : 6
model           : 54
model name      : Intel(R) Atom(TM) CPU N2800   @ 1.86GHz
stepping        : 1
microcode       : 0x10d
cpu MHz         : 798.000
cache size      : 512 KB
physical id     : 0
siblings        : 4
core id         : 0
cpu cores       : 2
apicid          : 0
initial apicid  : 0
fpu             : yes
fpu_exception   : yes
cpuid level     : 10
wp              : yes
flags           : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx lm constant_tsc arch_perfmon pebs bts nopl nonstop_tsc aperfmperf pni dtes64 monitor ds_cpl est tm2 ssse3 cx16 xtpr pdcm movbe lahf_lm arat dtherm
bogomips        : 3733.17
clflush size    : 64
cache_alignment : 64
address sizes   : 36 bits physical, 48 bits virtual
power management:
```

I didn't change anything "significant", except upgrades to last stable version (jessie).
But I think OVH pushed a custom linux kernel on their servers...
```
# uname -r
3.14.27-xxxx-grs-ipv6-64
```

Until now, I've never been worried about this kernel. Maybe something is missing or disabled in mine ?

**Comment by kouvel** [comment]
I couldn't find an older kernel to try in the repo. Would it be possible to try the current stable kernel for Jessie and switch back to the other kernel, to see if that has something to do with it?

This is what I have installed: https://packages.debian.org/jessie/linux-image-3.16.0-4-amd64

**Comment by geeooff** [comment]
@kouvel I will do it as soon as I can (need to change grub boot order, and reboot the server), but for now I must sleep a bit ;)
My other server, virtual one this time, still from OVH, where the same app was used to run successfully, was using the stock Ubuntu 16.04 LTS kernel :
```
# uname -r
4.4.0-47-generic
```

**Comment by janvorli** [comment]
@geeooff could you please repeat the experiment with running under gdb and setting breakpoint to ClrVirtualAllocWithinRange and when it gets hit, dump the stack trace using `bt` command for me? I'd like to see where this call came from. There are multiple code paths that can result in this function invocation.

**Comment by geeooff** [comment]
@janvorli Here's the [output](https://dl.dropboxusercontent.com/u/7169701/GitHub/dotnet/gdb_2016-11-25_callstack.log).
@kouvel I didn't forget you, i'm applying the generic kernel as soon as I can ;)

**Comment by geeooff** [comment]
@gkhanna79 @janvorli @kouvel My server is now running on the Debian 8.6 stable generic kernel, so without Grsecurity patches, and 3 instances of my app are running fine. I tried using GDB and `ClrVirtualAllocWithinRange` breakpoint wasn't hit at all with Server GC, only segfaults (expected ?) and no crash. Virtual Memory usage is still very high (approx 22 GB per instance).

I didn't check every page of my app, nor a load test, but everything seems okay now.

Perhaps it would be useful to additional tests with that kind of linux kernel (SELinux / AppArmor too ?) ? I don't know if they are common. Debian [have a wiki page on grsecurity](https://wiki.debian.org/grsecurity), and a kernel package for _stretch_ (next stable version).

Anyway, many thanks for reopening this issue, it helped me a lot, and I hope it will help .NET Core team too.

**Comment by janvorli** [comment]
@geeooff thank you for the details, I will definitely look into the grsecurity stuff, try to enable it on my local Debian 8.6 and see what we can do about it.
SEGFAULTS visible under the debugger most likely mean null references and if it runs ok without GDB, then they seem properly handled.
Thanks to the call stack you have provided us with, I can see why the ClrVirtualAllocWithinRange is called. Since we have failed to allocate memory for the executable memory allocator close to the libcoreclr.so in the memory, jitter needs to create jump stubs that can jump farther than +/-2GB. That's what results in calling ClrVirtualAllocWithinRange. But these jump stubs need to be located within +/-2GB range from the code that wants to make the long jump and we fail again, so the end result is OOM exception due to inability to allocate virtual memory in the specific range.
I guess that it has something to do with the grsecurity.

**Comment by janvorli** [comment]
@kamilosxd678 do you also happen to have grsecurity enabled on your Debian?

**Comment by thewavelength** [comment]
@janvorli Same issue here. Running the Hello-World-App works fine. Compiling also works fine. But running my project does not work (neither with the source code available and `dotnet run` or using `dotnet mydll.dll`).

Do you need any further debugging information?

uname: `Linux faraway 3.16.0-4-amd64 #1 SMP Debian 3.16.36-1+deb8u2 (2016-10-19) x86_64 GNU/Linux`
dotnet version: `1.0.0-preview2-1-003177` (I need 1.0.0 because of a MySQL EF driver dependency)

**Comment by janvorli** [comment]
@thewavelength if you could share the strace log for the failure, I could quickly check if it is the same or a different issue than the one we have discused here before.

**Comment by thewavelength** [comment]
@janvorli This is coming `dotnet MyProject.dll` (using ASP.NET Core):
[strace.txt](https://github.com/dotnet/cli/files/646046/strace.txt)

Running `dotnet new -t Web`, restoring it and then do `dotnet run` works as expected:
```
testuser@faraway:~/test$ dotnet run
Project test (.NETCoreApp,Version=v1.1) was previously compiled. Skipping compilation.
info: Microsoft.Extensions.DependencyInjection.DataProtectionServices[0]
      User profile is available. Using '/home/testuser/.aspnet/DataProtection-Keys' as key repository; keys will not be encrypted at rest.
Hosting environment: Production
Content root path: /home/testuser/test
Now listening on: http://localhost:5000
Application started. Press Ctrl+C to shut down.
```

**Comment by janvorli** [comment]
@thewavelength this is actually a different issue than the one discussed here before. In your case, the error code is 0x80131500. This error means a managed exception has occured during the initialization.
Looking at the strace, I can see it was trying to open libicuuc.so.55, but it hasn't found one anywhere. Could you please check what version of that library you have installed?

**Comment by thewavelength** [comment]
@janvorli Thanks.  Seems like I only have installed version 52. On some forums on the internet it says that Jessie only ships version 52. Why does the `dotnet runtime` require version 55 when selecting the version for Debian Jessie?

I didn't find a way to update this library to the latest version on Jessie.

```
/usr/lib/x86_64-linux-gnu/libicudata.a
/usr/lib/x86_64-linux-gnu/libicudata.so
/usr/lib/x86_64-linux-gnu/libicudata.so.52
/usr/lib/x86_64-linux-gnu/libicudata.so.52.1
/usr/lib/x86_64-linux-gnu/libicui18n.a
/usr/lib/x86_64-linux-gnu/libicui18n.so
/usr/lib/x86_64-linux-gnu/libicui18n.so.52
/usr/lib/x86_64-linux-gnu/libicui18n.so.52.1
/usr/lib/x86_64-linux-gnu/libicuio.a
/usr/lib/x86_64-linux-gnu/libicuio.so
/usr/lib/x86_64-linux-gnu/libicuio.so.52
/usr/lib/x86_64-linux-gnu/libicuio.so.52.1
/usr/lib/x86_64-linux-gnu/libicule.a
/usr/lib/x86_64-linux-gnu/libicule.so
/usr/lib/x86_64-linux-gnu/libicule.so.52
/usr/lib/x86_64-linux-gnu/libicule.so.52.1
/usr/lib/x86_64-linux-gnu/libiculx.a
/usr/lib/x86_64-linux-gnu/libiculx.so
/usr/lib/x86_64-linux-gnu/libiculx.so.52
/usr/lib/x86_64-linux-gnu/libiculx.so.52.1
/usr/lib/x86_64-linux-gnu/libicutest.a
/usr/lib/x86_64-linux-gnu/libicutest.so
/usr/lib/x86_64-linux-gnu/libicutest.so.52
/usr/lib/x86_64-linux-gnu/libicutest.so.52.1
/usr/lib/x86_64-linux-gnu/libicutu.a
/usr/lib/x86_64-linux-gnu/libicutu.so
/usr/lib/x86_64-linux-gnu/libicutu.so.52
/usr/lib/x86_64-linux-gnu/libicutu.so.52.1
/usr/lib/x86_64-linux-gnu/libicuuc.a
/usr/lib/x86_64-linux-gnu/libicuuc.so
/usr/lib/x86_64-linux-gnu/libicuuc.so.52
/usr/lib/x86_64-linux-gnu/libicuuc.so.52.1
```

**Comment by janvorli** [comment]
@thewavelength we should be building be package for the default libicu installed on each distro. It would then work on future versions. Let me check what do we set as the minimum version in the package for Jessie. Btw, where exactly did you get the dotnet package from and what version do you have?
In the meanwhile, you can use the following workaround. Please set the environment variable  CLR_ICU_VERSION_OVERRIDE to 52, that should fix the problem.

**Comment by thewavelength** [comment]
@janvorli I jumped to https://www.microsoft.com/net/download/linux and selected the x64 runtime for Debian (https://go.microsoft.com/fwlink/?LinkID=836295). That's all I did.

Setting the environment variable doesn't help. `strace` of the same command looks exactly the same and tries to search version 55.

```
myuser@mysystem:~/publish/myproject$ export CLR_ICU_VERSION_OVERRIDE=52
myuser@mysystem:~/publish/myproject$ dotnet myproject.dll
Failed to initialize CoreCLR, HRESULT: 0x80131500
myuser@mysystem:~/publish/myproject$ CLR_ICU_VERSION_OVERRIDE=52
myuser@mysystem:~/publish/myproject$ dotnet myproject.dll
Failed to initialize CoreCLR, HRESULT: 0x80131500
```

**Comment by janvorli** [comment]
@thewavelength ah, I see, that version of dotnet doesn't have the mechanism to try to get the version of libicu that's available on the box and only tries to load the version with which it was compiled. 

But, I have downloaded the package from the same link and I don't see it referring to libicu 55, but the libicu 52 as expected. 
Could you please run
`ldd /opt/dotnet/shared/Microsoft.NETCore.App/1.0.1/System.Globalization.Native.so` and check the dependencies?

**Comment by thewavelength** [comment]
@janvorli Okay, crazy. Here is the output of ldd:

```
linux-vdso.so.1 (0x00007fff6abf6000)
libicuuc.so.55 => not found
libicui18n.so.55 => not found
libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f3f1a941000)
libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f3f1a63d000)
libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f3f1a426000)
libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f3f1a088000)
/lib64/ld-linux-x86-64.so.2 (0x00007f3f1aed0000)
```

**Comment by janvorli** [comment]
This is really strange. The only explanation I have is that you could have accidentally downloaded a package for another Linux. Can you try to download the correct package again, untar it to some local folder and run ldd on the shared/Microsoft.NETCore.App/1.0.1/System.Globalization.Native.so in it?

**Comment by janvorli** [comment]
The link for the one for Ubuntu 16.04 is right above it and it refers to libicu 55.

**Comment by thewavelength** [comment]
@janvorli I'm going to sh** bricks... I fully cleaned anything that related to dotnet on my server and following exactly the instructions on the microsoft page with the 1.0.1-runtime (https://www.microsoft.com/net/core#linuxdebian). And it just works. What. The. Fuck.
What I don't get: if I now do the `ldd` command to the path you've mentioned, it states it does not exists.

I really can't reproduce what went wrong in the past but probably I just collected the wrong download URL and wasted up my server. Sorry for your lost time!

```
root@faraway:/home/main# rm /usr/local/bin/dotnet
root@faraway:/home/main# rm -rf /opt/dotnet
root@faraway:/home/main# cd /tmp
root@faraway:/tmp# dotnet
bash: /usr/local/bin/dotnet: No such file or directory
root@faraway:/tmp# mkdir installdotnet
root@faraway:/tmp# cd installdotnet
root@faraway:/tmp/installdotnet# curl -sSL -o dotnet.tar.gz https://go.microsoft.com/fwlink/?LinkID=836295
root@faraway:/tmp/installdotnet# mkdir -p /opt/dotnet
root@faraway:/tmp/installdotnet# tar zxf dotnet.tar.gz -C /opt/dotnet
root@faraway:/tmp/installdotnet# ln -s /opt/dotnet/dotnet /usr/local/bin
root@faraway:/tmp/installdotnet# dotnet --version

Microsoft .NET Core Shared Framework Host

  Version  : 1.0.1
root@faraway:/tmp/installdotnet# ldd /opt/dotnet/shared/Microsoft.NETCore.App/1.0.1/System.Globalization.Native.so
ldd: /opt/dotnet/shared/Microsoft.NETCore.App/1.0.1/System.Globalization.Native.so: No such file or directory
root@faraway:/tmp/installdotnet# su myuser
sander@faraway:~$ dotnet myproject.dll
```

**Comment by janvorli** [comment]
No  problem, I am glad it works for you now :-)

**Comment by thewavelength** [comment]
@janvorli Thanks for your effort. Great work with dotnet core :-) I really like it!

**Comment by gkhanna79** [comment]
Awesome that this is resolved now :) @thewavelength Is the issue good to close?

**Comment by thewavelength** [comment]
@gkhanna79 Yes, please proceed. But please note I'm not the original poster and therefore can't decide for him :-)

**Comment by gkhanna79** [comment]
@prafullbhosale Any inputs?

**Comment by prafullbhosale** [comment]
@gkhanna79 Yes, good to close. 
I had originally opened this issue for windows, but recently starting the same issue as @thewavelength on Ubuntu. 
Thanks @janvorli  and @thewavelength for the workarounds!!

**Comment by markvincze** [comment]
I'm getting now the same error after upgrading my API to 2.0.

On my local machine it works properly, but it fails with this if I try to run it in Docker with the base `microsoft/dotnet:2.0.0-runtime`.  
Is this familiar to anyone?

My `TargetFramework` is `netcoreapp2.0`, and my `RuntimeFrameworkVersion` is `2.0.0`.

**Comment by markvincze** [comment]
Hmm, it turns out if I switch from the `debian.8-x65` runtime to `linux-x64`, then it works.  
Btw. what does the `linux-x64` RID represent exactly? I found it in one of the examples repository, but I cannot find it on the page listing all the RIDs.

**Comment by janvorli** [comment]
@markvincze the `linux-x64` RID represents all reasonably recent x64 Linux distros that use glibc as the standard C++ library. The RHEL 7 is the oldest distro supported.
